src/App.js

```diff
@@ -1,11 +1,124 @@
+      .then(resp => {
+        // Only return popular movies whos IDs are NOT found in my savedMovies
+        this.setState({
+          movies: resp.data.results.filter(resultsMovie =>
+            !savedMoviesIdArr[savedMoviesIdArr.indexOf(resultsMovie.id)])
+        });
+      });
+  }
+  handleSearchBarChange(event) {
+    this.setState({
+      searchText: event.target.value
+    });
+  }
+  handleSubmit(event) {
+    event.preventDefault();
+    const savedMoviesIdArr = this.state.savedMovies.map(savedMovie => savedMovie.id);
+    axios.get(`https://api.themoviedb.org/3/search/movie?api_key=2dba200e2682e0f8903ed87b9c9e02d1&language=en-US&query=${this.state.searchText}&page=1&include_adult=false`)
+      .then(resp => {
+        this.setState({
+          // Only return searched movies whos IDs are NOT found in my savedMovies
+          movies: resp.data.results.filter(resultsMovie =>
+            !savedMoviesIdArr[savedMoviesIdArr.indexOf(resultsMovie.id)])
+        });
+      })
+      .catch(err => {
+        console.err(`Error ${err}`);
+      });
+  }
+  handleImgError(event) {
+    event.target.src = 'http://i.imgur.com/SUynOc5.png';
+  }
+  rocketFaveHandle(event, id) {
+    event.preventDefault();
+    const movieIdArr = (this.state.movies.map(movieItem => {
+      return movieItem.id;
+    }));
+    const savedMovieIdArr = (this.state.savedMovies.map(savedMovie => {
+      return savedMovie.id;
+    }));
+    if (savedMovieIdArr.indexOf(id) >= 0) {
+      // This movie ID exists in my saved movies ID array, so DELETE it from my db.json
+      axios.delete(`http://localhost:4000/movies/${this.state.savedMovies[movieIdArr.indexOf(id)]._id}`)
+          .then(resp => {
+            // Remove movie object from both states (from the page and savedMovies)
+            const newMovies = this.state.movies.filter(movie => movie.id !==
+              this.state.movies[movieIdArr.indexOf(id)].id);
+            const newSavedMovies = this.state.savedMovies.filter(movie => movie.id !==
+              (this.state.movies[movieIdArr.indexOf(id)].id));
+            this.setState({
+              movies: newMovies,
+              savedMovies: newSavedMovies
+            });
+          });
+    } else {
+      // This movie ID is not in my saved movies ID array, so POST it to my db.josn
+        axios.post('http://localhost:4000/movies', (this.state.movies[movieIdArr.indexOf(id)]))
+          .then(resp => {
+            // Remove movie object from movie state (from the page) and
+            // concat it to savedMovies state
+            const newMovies = this.state.movies.filter(movie => movie.id !==
+              this.state.movies[movieIdArr.indexOf(id)].id);
+            this.setState({
+              movies: newMovies,
+              savedMovies: [...this.state.savedMovies, resp.data]
+            });
+          });
+    }
+  }
+  showRocketFavs() {
```

Comment:
I think this idea is moving in the right direction, but we've introduced duplicate data into our state.

It would be better in this case to say, have a `boolean` , something like `showFavoriteMovies` that we can toggle true or false. This way our movies only ever have _one source of truth_